### PR TITLE
chore(release): update 3.0.0 yamls

### PR DESCRIPTION
**Pull request by kmova**
#### Why is this change required?
Updating the 3.0.0 YAMLs for various engines. 

#### What is changed?
Other than updating the yamls with latest version, this PR also is trying to do the following:
- Add scripts that can pull in the engine specific yamls and help with updating the release tags. 
- Add script to generate the openebs-operator from individual yamls. 
- Cleanup or move the deprecated yamls 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/openebs/charts/blob/HEAD/CONTRIBUTING.md#sign-your-commits) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

**Review by mittachaitu — COMMENTED** [review]
reviewed cStor and dynamic-nfs-provisioner artifacts, looks like we have normalized artifacts in individual repos(cstor-operator, jiva-operator and node-disk-manager)

File: yamls/local-hostpath-provisioner.yaml [hunk 1]
```diff
@@ -0,0 +1,133 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: openebs-localpv-provisioner
+  namespace: openebs
+  labels:
+    name: openebs-localpv-provisioner
+    openebs.io/component-name: openebs-localpv-provisioner
+    openebs.io/version: 2.12.2
```
Comment (niladrih) [inline yamls/local-hostpath-provisioner.yaml:9]: Label is of 2.12.2

File: yamls/local-hostpath-provisioner.yaml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      labels:
+        name: openebs-localpv-provisioner
+        openebs.io/component-name: openebs-localpv-provisioner
+        openebs.io/version: 2.12.2
+    spec:
+      serviceAccountName: openebs-maya-operator
+      containers:
+      - name: openebs-provisioner-hostpath
+        imagePullPolicy: IfNotPresent
+        image: openebs/provisioner-localpv:3.0.0
```
Comment (niladrih) [inline yamls/local-hostpath-provisioner.yaml:29]: Label and image version do not match.

**Review by niladrih — COMMENTED** [review]
Left few comments.

File: versioned/3.0.0/nfs-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      openebs.io/component-name: openebs-nfs-provisioner
+  replicas: 1
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        name: openebs-nfs-provisioner
+        openebs.io/component-name: openebs-nfs-provisioner
+        openebs.io/version: 0.7.1
```
Comment (mynktl) [inline versioned/3.0.0/nfs-operator.yaml:87]: should this be 3.0.0 or a component version?

File: cstor-operator.yaml [hunk 4]
```diff
@@ -362,3908 +285,4331 @@ status:
...
-                                type: object
-                              type: array
-                            requiredDuringSchedulingIgnoredDuringExecution:
-                              description: If the affinity requirements specified by
-                                this field are not met at scheduling time, the pod will
-                                not be scheduled onto the node. If the affinity requirements
-                                specified by this field cease to be met at some point
-                                during pod execution (e.g. due to a pod label update),
-                                the system may or may not try to eventually evict the
-                                pod from its node. When there are multiple elements,
```
[the remark this answers is not in the document]
Comment (nsathyaseelan) [reply cstor-operator.yaml:3780]: and also daemonset name is changed form `openebs-ndm` into n`ode-disk-manager` is this expected change?

Comment (akhilerm) [inline cstor-operator.yaml:3810]:
```suggestion
        #- --feature-gates="APIService"
```

Comment (akhilerm) [inline cstor-operator.yaml:3815]:
```suggestion
        #- --feature-gates="UseOSDisk"
```

Comment (akhilerm) [inline cstor-operator.yaml:3857]:
```suggestion
          value: "0"
```

We no longer need to create a sparse file by default

File: microk8s-cstor-operator.yaml [hunk 5]
```diff
@@ -1,3 +1,9 @@
+# This manifest is autogenerated via 'make manifests' command.
+# Do the modification to the yamls in deploy/yamls/  directory
+# and then run 'make manifests' command
+#
+# NDM Operator YAML will be downladed from github.com/openebs/node-disk-manager repo
```
Comment (akhilerm) [inline microk8s-cstor-operator.yaml]: is this done as part of `make manifests`?

File: microk8s-cstor-operator.yaml [hunk 6]
```diff
@@ -362,3908 +285,4331 @@ status:
...
-                                  - weight
-                                type: object
-                              type: array
-                            requiredDuringSchedulingIgnoredDuringExecution:
-                              description: If the affinity requirements specified by
-                                this field are not met at scheduling time, the pod will
-                                not be scheduled onto the node. If the affinity requirements
-                                specified by this field cease to be met at some point
-                                during pod execution (e.g. due to a pod label update),
-                                the system may
```
Comment (akhilerm) [inline microk8s-cstor-operator.yaml]:
```suggestion
        #- --feature-gates="APIService"
        # Detect changes to device size, filesystem and mount-points without restart.
        #- --feature-gates="ChangeDetection"
        # Default address is 0.0.0.0:9115, do not use quotes around the address
        # - --api-service-address=0.0.0.0:9115
        #- --feature-gates="UseOSDisk"
```

Comment (akhilerm) [inline microk8s-cstor-operator.yaml]:
```suggestion
        #- --feature-gates="APIService"
```

Comment (akhilerm) [inline microk8s-cstor-operator.yaml]:
```suggestion
        #- --feature-gates="APIService"
```

Comment (akhilerm) [inline microk8s-cstor-operator.yaml]:
```suggestion
        #- --feature-gates="UseOSDisk"
```

File: openebs-operator.yaml [hunk 7]
```diff
@@ -897,10 +955,10 @@ spec:
...
       - name: node-disk-manager
-        image: openebs/node-disk-manager:1.6.1
+        image: openebs/node-disk-manager:1.7.0
         args:
           - -v=4
-          # The feature-gate is used to enable the new UUID algorithm.
+        # The feature-gate is used to enable the new UUID algorithm.
           - --feature-gates="GPTBasedUUID"
         # Detect mount point and filesystem changes wihtout restart.
         # Uncomment the line below to enable the feature.
```
Comment (akhilerm) [inline openebs-operator.yaml]: This needs to be changed and updated with ChangeDetection in the comments

File: versioned/3.0.0/cstor-operator.yaml [hunk 8]
```diff
@@ -0,0 +1,5576 @@
...
+        image: openebs/node-disk-manager:1.7.0
+        args:
+        - -v=4
+        - --feature-gates="GPTBasedUUID"
+        - --feature-gates="APIService"
+        # Detect changes to device size, filesystem and mount-points without restart.
+        #- --feature-gates="ChangeDetection"
+        # Default address is 0.0.0.0:9115, do not use quotes around the address
+        # - --api-service-address=0.0.0.0:9115
+        - --feature-gates="UseOSDisk"
```
Comment (akhilerm) [inline versioned/3.0.0/cstor-operator.yaml]:
```suggestion
        #- --feature-gates="APIService"
        # Detect changes to device size, filesystem and mount-points without restart.
        #- --feature-gates="ChangeDetection"
        # Default address is 0.0.0.0:9115, do not use quotes around the address
        # - --api-service-address=0.0.0.0:9115
        #- --feature-gates="UseOSDisk"
```

File: versioned/3.0.0/microk8s-cstor-operator.yaml [hunk 9]
```diff
@@ -0,0 +1,5576 @@
...
+        image: openebs/node-disk-manager:1.7.0
+        args:
+        - -v=4
+        - --feature-gates="GPTBasedUUID"
+        - --feature-gates="APIService"
+        # Detect changes to device size, filesystem and mount-points without restart.
+        #- --feature-gates="ChangeDetection"
+        # Default address is 0.0.0.0:9115, do not use quotes around the address
+        # - --api-service-address=0.0.0.0:9115
+        - --feature-gates="UseOSDisk"
```
Comment (akhilerm) [inline versioned/3.0.0/microk8s-cstor-operator.yaml]:
```suggestion
        #- --feature-gates="APIService"
        # Detect changes to device size, filesystem and mount-points without restart.
        #- --feature-gates="ChangeDetection"
        # Default address is 0.0.0.0:9115, do not use quotes around the address
        # - --api-service-address=0.0.0.0:9115
        #- --feature-gates="UseOSDisk"
```

File: versioned/3.0.0/openebs-operator.yaml [hunk 10]
```diff
@@ -0,0 +1,1374 @@
...
+      containers:
+      - name: node-disk-manager
+        image: openebs/node-disk-manager:1.7.0
+        args:
+          - -v=4
+        # The feature-gate is used to enable the new UUID algorithm.
+          - --feature-gates="GPTBasedUUID"
+        # Detect mount point and filesystem changes wihtout restart.
+        # Uncomment the line below to enable the feature.
+        # --feature-gates="MountChangeDetection"
```
Comment (akhilerm) [inline versioned/3.0.0/openebs-operator.yaml]: Change this to ChangeDetection

File: yamls/ndm-operator.yaml [hunk 11]
```diff
@@ -461,7 +461,7 @@ spec:
       #hostPID: true
       containers:
       - name: node-disk-manager
-        image: openebs/node-disk-manager:1.6.1
+        image: openebs/node-disk-manager:1.7.0
         args:
           - -v=4
         # The feature-gate is used to enable the new UUID algorithm.
```
Comment (akhilerm) [inline yamls/ndm-operator.yaml]: The flag needs to be renamed to ChangeDetection on [line](https://github.com/openebs/charts/pull/282/files?file-filters%5B%5D=.yaml&hide-deleted-files=true#diff-3669f197ed8f059697b8c8e73444abb5dbff592666cb33180040cc9a439c835aR471)

**Review by akhilerm — CHANGES_REQUESTED** [review]
suggested a few changes

File: versioned/3.0.0/nfs-operator.yaml (hunk 3 above)
Comment (kmova) [reply versioned/3.0.0/nfs-operator.yaml:87]: this is ok to be on the engine version.

File: versioned/3.0.0/nfs-operator.yaml [hunk 12]
```diff
@@ -0,0 +1,214 @@
+# This manifest deploys the OpenEBS control plane components, with associated CRs & RBAC rules
```
[the remark this answers is not in the document]
Comment (kmova) [reply versioned/3.0.0/nfs-operator.yaml:1]: was already checked in into parent directory in previous PR. Was missed under 3.0.0

File: scripts/release/fetch-latest-yaml.sh [hunk 13]
```diff
@@ -57,9 +57,13 @@ fetch_engine_operator_yaml()
     echo "Success: Downloaded ${C_URL} into ${TEMP_RESP_FILE}"
   fi
 
+  sed -e 's@node-disk-manager:ci@node-disk-manager:1.7.0@' -i ${TEMP_RESP_FILE}
+  sed -e 's@node-disk-operator:ci@node-disk-operator:1.7.0@' -i ${TEMP_RESP_FILE}
+  sed -e 's@node-disk-exporter:ci@node-disk-exporter:1.7.0@' -i ${TEMP_RESP_FILE}
   sed -e 's@\:ci@'"\:$TAG"'@' -i ${TEMP_RESP_FILE}
```
Comment (akhilerm) [inline scripts/release/fetch-latest-yaml.sh]: Do we also need to change the linux-utils image to 3.0.0?

Comment (akhilerm) [reply scripts/release/fetch-latest-yaml.sh]: Also, when is this script run?

**Review by akhilerm — COMMENTED** [review]
NDM related YAMLs looks good. Have one comment on the script used to generate the YAMLs.

File: ndm-operator.yaml [hunk 14]
```diff
@@ -666,49 +724,56 @@ spec:
 apiVersion: v1
 kind: Service
 metadata:
-  name: ndm-cluster-exporter-service
+  name: openebs-ndm-cluster-exporter-service
   namespace: openebs
   labels:
-    name: openebs-ndm-exporter
-    component: openebs-ndm-cluster-exporter
+    name: openebs-ndm-clsuter-exporter-service
```
Comment (Ab-hishek) [inline ndm-operator.yaml]: This was the common label used for monitoring @kmova which we decided to keep as `app: openebs-ndm-exporter`. The changes are there in https://github.com/openebs/node-disk-manager/pull/647 but not merged.

File: ndm-operator.yaml [hunk 15]
```diff
@@ -731,17 +796,17 @@ spec:
 apiVersion: v1
 kind: Service
 metadata:
-  name: ndm-node-exporter-service
+  name: openebs-ndm-node-exporter-service
   namespace: openebs
   labels:
-    name: openebs-ndm-exporter
-    component: openebs-ndm-node-exporter
+    name: openebs-ndm-node-exporter
```
Comment (Ab-hishek) [inline ndm-operator.yaml]: Same comment as above https://github.com/openebs/charts/pull/282/files#r714493358

File: openebs-operator-lite.yaml [hunk 16]
```diff
@@ -680,6 +679,149 @@ spec:
...
+# Create NDM cluster exporter service
+# This is optional and required only when
+# ndm-cluster-exporter deployment is used
+apiVersion: v1
+kind: Service
+metadata:
+  name: openebs-ndm-cluster-exporter-service
+  namespace: openebs
+  labels:
+    name: openebs-ndm-clsuter-exporter-service
```
Comment (Ab-hishek) [inline openebs-operator-lite.yaml]: Same label comment here also.

Comment (Ab-hishek) [reply openebs-operator-lite.yaml]: And in the corresponding versioned ndm-operator file.

**Review by mittachaitu — COMMENTED** [review]
We need to incorporate below changes in cstor-operators repo, right?

File: scripts/release/fetch-latest-yaml.sh (hunk 13 above)
Comment (kmova) [reply scripts/release/fetch-latest-yaml.sh]: This script is not yet used. Currently used to find the differences across multiple repos and have identified some and fixed in this PR.

File: jiva-operator.yaml [hunk 17]
```diff
@@ -3158,262 +3166,7 @@ status:
     plural: ""
   conditions: []
   storedVersions: []
----
```
Comment (shubham14bajpai) [inline jiva-operator.yaml:3161]: Upgradetask crd is still required.

File: jiva-operator.yaml [hunk 18]
```diff
@@ -3552,9 +3305,9 @@ spec:
             - name: OPERATOR_NAME
               value: "jiva-operator"
             - name: OPENEBS_IO_JIVA_CONTROLLER_IMAGE
-              value: "openebs/jiva:2.12.2"
+              value: "openebs/jiva:3.0.0"
             - name: OPENEBS_IO_JIVA_REPLICA_IMAGE
-              value: "openebs/jiva:2.12.2"
+              value: "openebs/jiva:3.0.0"
```
Comment (shubham14bajpai) [inline jiva-operator.yaml:3310]:
```suggestion
              value: "openebs/jiva:3.0.0"
            - name: OPENEBS_IO_MAYA_EXPORTER_IMAGE
              value: "openebs/m-exporter:3.0.0"
```

File: versioned/3.0.0/jiva-operator.yaml [hunk 19]
```diff
@@ -0,0 +1,3793 @@
...
+            - name: OPENEBS_SERVICEACCOUNT_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.serviceAccountName
+            - name: OPERATOR_NAME
+              value: "jiva-operator"
+            - name: OPENEBS_IO_JIVA_CONTROLLER_IMAGE
+              value: "openebs/jiva:3.0.0"
+            - name: OPENEBS_IO_JIVA_REPLICA_IMAGE
+              value: "openebs/jiva:3.0.0"
```
Comment (shubham14bajpai) [inline versioned/3.0.0/jiva-operator.yaml]:
```suggestion
              value: "openebs/jiva:3.0.0"
            - name: OPENEBS_IO_MAYA_EXPORTER_IMAGE
              value: "openebs/m-exporter:3.0.0"
```

File: versioned/3.0.0/jiva-operator.yaml [hunk 20]
```diff
@@ -0,0 +1,3793 @@
+
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openebs
+
+---
```
Comment (shubham14bajpai) [inline versioned/3.0.0/jiva-operator.yaml]: Need to add upgradetask CRD

**Review by akhilerm — APPROVED** [review]
LGTM

**Review by shubham14bajpai — APPROVED** [review]
jiva-operator looks good :+1:

**Review by mittachaitu — APPROVED** [review]
LGTM

**Review by Ab-hishek — APPROVED** [review]
ndm operator changes looks good.

**Review by niladrih — APPROVED** [review]

**Pull request merged**